CRM/Core/Permission/Joomla.php

```diff
@@ -131,4 +135,74 @@ public function checkGroupRole($array) {
     return FALSE;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function upgradePermissions($permissions) {
+    $translatedPerms = array();
+    foreach (array_flip($permissions) as $perm) {
+      $translated = $this->translateJoomlaPermission($perm);
+      $translatedPerms[] = $translated[0];
+    }
+
+    $associations = $this->getUserGroupPermsAssociations();
+    $originalAssociations = clone $associations;
+    foreach (array_keys(get_object_vars($associations)) as $permName) {
+      if (!in_array($permName, $translatedPerms)) {
+        unset($associations->$permName);
+      }
+    }
+
+    if ($originalAssociations !== $associations) {
```

Comment:
I think you should just set $dirty = true in the loop.

I _think_ the non-class-sensitive boolean operator would have the intended behaviour, as the the docs say that that properties and values are compared.  But I don't think it is a good idiom and you are exchanging CPU cycles (comparing every property) for a few keystrokes, seems like a bad bargain.